### PR TITLE
TNO-270: Add ability to delete data source

### PR DESCRIPTION
**Pull request by jdtoombs**
_Styling obviously not yet completed._

deleteDataSourceTakeTwo

### CODE REVIEW

File: app/editor/src/components/form/formpage/styled/FormPage.tsx [hunk 1]
```diff
@@ -4,7 +4,7 @@ import { IFormPageProps } from '../FormPage';
 
 export const FormPage = styled.div<IFormPageProps>`
   background-color: white;
-  height: ${(props) => props.height ?? '100%'};
+  min-height: 100%;
```
Comment (jdtoombs) [inline app/editor/src/components/form/formpage/styled/FormPage.tsx:7]: Unrelated to story, but so the `FormPage` will always take up full space of vertical as indicated in mockups

File: app/editor/src/features/content/form/ContentForm.tsx [hunk 2]
```diff
@@ -102,7 +102,7 @@ export const ContentForm: React.FC = () => {
 
   return (
     <styled.ContentForm>
-      <FormPage height="fit-content">
+      <FormPage>
```
Comment (jdtoombs) [inline app/editor/src/features/content/form/ContentForm.tsx:105]: No longer needed with `min-height` change indicated at top of PR

**Review by Fosol — APPROVED** [review]

File: app/editor/src/features/admin/sources/DataSource.tsx [hunk 3]
```diff
@@ -66,9 +70,28 @@ export const DataSource: React.FC<IDataSourceProps> = (props) => {
...
+            <Modal
+              headerText="Confirm Removal"
+              body="Are you sure you wish to remove this data source?"
+              isShowing={isShowing}
+              hide={toggle}
+              type="delete"
+              confirmText="Yes, Remove It"
+              onConfirm={async () => {
+                try {
+                  await api.deleteDataSource(source);
```
Comment (Fosol) [inline app/editor/src/features/admin/sources/DataSource.tsx:88]: The data sources are in redux memory, this won't delete the item from the in-memory collection.

**Review by Fosol — CHANGES_REQUESTED** [review]
Need to handle the redux memory issue.

File: app/editor/src/features/admin/sources/DataSource.tsx (hunk 3 above)
Comment (jdtoombs) [reply app/editor/src/features/admin/sources/DataSource.tsx:88]: Just to verify, does this line not cover that?
`store.storeDataSources(getSources().filter((ds) => ds.id !== result.id));`

Comment (jdtoombs) [reply app/editor/src/features/admin/sources/DataSource.tsx:88]: found within the `deleteDataSource` method

Comment (Fosol) [reply app/editor/src/features/admin/sources/DataSource.tsx:88]: It probably does!

**Review by Fosol — APPROVED** [review]

**Pull request merged**